keps/sig-cluster-lifecycle/kubeadm/1739-customization-with-patches/README.md

```diff
@@ -424,6 +433,28 @@ but also allow for advanced usage with patch ordering and the `json` type.
 - Would allow having arbitrary files that are ignored in the patch folder -
 e.g. having a `README.md`.
 
+### Patch targets
+
+The mechanism allows patching the following targets via the `target` naming property:
+
+#### Control plane static Pods
+
+Static Pod manifests generated on control plane nodes can be patched with the following
+targets: `kube-apiserver`, `kube-controller-manager`, `kube-scheduler` and `etcd`.
+Their API GroupVersionKind is `core/v1.Pod`.
+
+Patches for the control plane static Pods are applied over the generated manifests
+in memory during `init/join/upgrade`.
+
+#### Kubelet ComponentConfig
+
+The generated per-node kubelet ComponentConfig can be patched with the `kubeletconfiguration`
+target. Its API GroupVersionKind is of type `kubelet.config.k8s.io/v1beta1.KubeletConfiguration`.
+
+During `init` patches for the kubelet are applied over the user provided and kubeadm defaulted
+`KubeletConfiguration`. During `join/upgrade` patches are applied over the downloaded
+from the cluster global `KubeletConfiguration`.
```

Comment:
The update LGTM. 
My only concern is about how to make a new user notice this option and there is a patch for some components.
- the documentation
- maybe we can add some tips in the configmap annotation like "kubeadm.kubernetes.io/patched.component=kube-apiserver".
- some logs during kubeadm `init/join/upgrade` process.

Reply:
I guess the presense of a --patches flag or the option in config is the main indicator that components are being patched. And the user technically owns / maintains these flags and the config, so they must know what they are passing to kubeadm.

> the documentation

I plan to update this page:
https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/control-plane-flags/

> maybe we can add some tips in the configmap annotation like "kubeadm.kubernetes.io/patched.component=kube-apiserver".

Today we do not annotate static pods whether they are modified from kubeadm defaults. There is this extra  customization layer due to extraArgs, volumes and mounts that applies over the defaults. We assume the config file is the source if truth and users will be able to trace back changes to it. Patches are also a customization layer and we assume that new users should look at the node config and see there is a patches directory.

I think such an annotation is doable, but its not clear to me whether it's going to be a net positive addition for us and the users.

> some logs during kubeadm init/join/upgrade process.

Yes, logs already print that a patch folder is being pricessed and what patches were found there.

Reply:
/lgtm